src/dotnet/SlnProjectExtensions.cs

```diff
@@ -12,17 +12,29 @@ public static class SlnProjectExtensions
     {
         public static IList<string> GetSolutionFoldersFromProject(this SlnProject project)
         {
-            var currentDirString = $".{Path.DirectorySeparatorChar}";
+            var solutionFolders = new List<string>();
 
-            var directoryPath = Path.GetDirectoryName(project.FilePath);
-            if (directoryPath.StartsWith(currentDirString))
+            var projectFilePath = project.FilePath;
+            if (!projectFilePath.StartsWith(".."))
```

Comment:
Because I don't want to add solution folders for things outside the directory structure rooted at the solution. @blackdwarf is that a valid assumption?